fakeredis/commands_mixins/hash_mixin.py

```diff
@@ -134,3 +138,136 @@ def hrandfield(self, key: CommandItem, *args: bytes) -> Optional[List[bytes]]:
+    def hexpire(
+        self,
+        key: CommandItem,
+        seconds: Union[int, datetime.timedelta],
+        *fields: bytes,
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> list[HexpireResult]:
+        return [self._set_key_expiration(key, field, seconds, False, nx, xx, gt, lt) for field in fields]
+
+    @command(name="HPEXPIRE", fixed=(Key, Union[float, datetime.timedelta]), repeat=(bytes,))
+    def hpexpire(
+        self,
+        key: CommandItem,
+        milliseconds: float,
+        *fields: bytes,
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> list[HexpireResult]:
+        return [self._set_key_expiration(key, field, milliseconds, True, nx, xx, gt, lt) for field in fields]
+
+    @command(name="HEXPIREAT", fixed=(Key, Union[int, datetime.datetime]), repeat=(bytes,))
+    def hexpireat(
+        self,
+        key: CommandItem,
+        unix_time_seconds: Union[int, datetime.datetime],
+        *fields: bytes,
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> list[HexpireResult]:
+        return [self._set_key_expiration(key, field, unix_time_seconds, False, nx, xx, gt, lt) for field in fields]
+
+    @command(name="HPEXPIREAT", fixed=(Key, Union[float, datetime.datetime]), repeat=(bytes,))
+    def hpexpireat(
+        self,
+        key: CommandItem,
+        unix_time_milliseconds: float,
+        *fields: bytes,
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> list[HexpireResult]:
+        return [self._set_key_expiration(key, field, unix_time_milliseconds, True, nx, xx, gt, lt) for field in fields]
+
+    @command(name="HPERSIST", fixed=(Key,), repeat=(bytes,))
+    def hpersist(self, key: CommandItem, *fields: bytes) -> list[int]:
+        hash_val: Hash = key.value
+        return [
+            -2 if field not in hash_val._expirations else (1 if hash_val._clear_expiration(field) else -1)
+            for field in fields
+        ]
+
+    @command(name="HEXPIRETIME", fixed=(Key,), repeat=(bytes,))
+    def hexpiretime(self, key: CommandItem, *fields: bytes) -> list[int]:
+        hash_val: Hash = key.value
+        return [
+            -2 if field not in hash_val._expirations else int(hash_val._get_expiration(field) or -1) for field in fields
+        ]
+
+    @command(name="HPEXPIRETIME", fixed=(Key,), repeat=(bytes,))
+    def hpexpiretime(self, key: CommandItem, *fields: bytes) -> list[float]:
+        hash_val: Hash = key.value
+        return [
+            -2 if field not in hash_val._expirations else float(hash_val._get_expiration(field) or -1)
+            for field in fields
+        ]
+
+    @command(name="HTTL", fixed=(Key,), repeat=(bytes,))
+    def httl(self, key: CommandItem, *fields: bytes) -> list[int]:
```

Comment:
python 3.8+ is supported, please use typings that are compatible with 3.8 (`List[int]`)